src/components/SearchShopByName.vue

```diff
@@ -0,0 +1,49 @@
+<template>
+    <div class="search-by-name">
+        <input type="text" v-model="query" v-on:input="showSearchResults()">
+        <shop-list :shops="shops" />
+   </div>
+</template>
+
+<script>
+import { requestSearchByName } from '@/io/io';
+import ShopList from '@/components/ShopList';
+
+export default
+{
+    name: 'search-shop-by-name',
+    data ()
+    {
+        return {
+            query: '',
+            shops: [],
+        };
+    },
+    methods:
+    {
+        showSearchResults ()
+        {
+            requestSearchByName(this.query)
+            .then(result =>
+            {
+                this.shops = result;
+            })
+            // todo: handle this better somehow
+            .catch(sth => sth);
```

Comment:
This is a stub catcher with a todo -> Network errors will be handled in pirateup/tasks#33